spec/models/post_mover_spec.rb

```diff
@@ -615,6 +615,30 @@ def create_topic_user(user, opts = {})
             expect(topic.closed).to eq(true)
           end
 
+          it "schedules topic deleting when all posts was moved" do
+            SiteSetting.days_to_wait_before_deleting_fully_merged_stub_topics = 7
+            freeze_time
+
+            topic.expects(:add_moderator_post).twice
+            moved_to = topic.move_posts(user, [p1.id, p2.id, p3.id, p4.id], destination_topic_id: destination_topic.id)
+            expect(moved_to).to be_present
+
+            timer = topic.topic_timers.detect { |t| t.status_type == TopicTimer.types[:delete] }
```

Comment:
The following should be a little bit more efficient.

```suggestion
            timer = topic.topic_timers.find_by(status_type: TopicTimer.types[:delete])
```

Reply:
I've applied this in a separate commit.